Deprecate Jaeger / OpenTracing based tracing support (#7195)
strimzi/strimzi-kafka-operator: documentation/assemblies/tracing/assembly-distributed-tracing.adoc

* MirrorMaker 2.0
* Strimzi Kafka Bridge

> [NOTE]

Reviewer: We haven't mentioned Jaeger yet, so maybe we should add a sentence before the note.

_By default, Strimzi uses Jaeger tracing. It is enabled by setting the `spec.tracing.type` property to `jaeger` in the component configuration._

====

> The `type: jaeger` tracing support based on Jaeger clients and OpenTracing API is deprecated.
> Because the Jaeger clients are retired and the OpenTracing project is archived, we cannot guarantee their support for future Kafka versions.
> In the future, we plan to replace it with a new tracing feature based on the OpenTelemetry project.

Reviewer: ```suggestion
Support for `type: jaeger` tracing is deprecated.
The Jaeger clients are now retired and the OpenTracing project archived. 
As such, we cannot guarantee their support for future Kafka versions.
In the future, we plan to introduce a new tracing feature based on the OpenTelemetry project.
```
